File: dash_docs/chapters/advanced_callbacks/examples/callbacks-initial-call.py
import dash
from dash.dependencies import Input, Output
import dash_html_components as html

app = dash.Dash()
app.layout = html.Div(
    [
        html.Button("execute callback", id="button_1"),
        html.Div(children="callback not executed", id="first_output_1"),
        html.Div(children="callback not executed", id="second_output_1"),
    ]
)


@app.callback(
    [Output("first_output_1", "children"), Output("second_output_1", "children")],
    [Input("button_1", "n_clicks")]
)
def change_text(n_clicks):
    return ["n_clicks is " + str(n_clicks), "n_clicks is " + str(n_clicks)]

if __name__ == '__main__':
    app.run_server(debug=True)

File: dash_docs/chapters/advanced_callbacks/examples/callbacks-initial-call.R
library(dash)
library(dashHtmlComponents)


app <- Dash$new()

app$layout(
    htmlDiv(
        list(
            htmlButton("execute callback", id="button_1"),
            htmlDiv("callback not executed", id="first_output_1"),
            htmlDiv("callback not executed", id="second_output_1")
        )
    )
)

app$callback(
    list(output("first_output_1", "children"),
         output("second_output_1", "children")),
    params=list(input("button_1", "n_clicks")),
    function(n_clicks) {
        return(list(paste0("n_clicks is ", as.character(n_clicks)),
                    paste0("n_clicks is ", as.character(n_clicks))
            )
        )
    }
)

app$run_server()
